### ISSUE TITLE
Error transpiling set into Rust

### ISSUE THREAD

**Issue by ajewellamz** [dafny-lang/dafny#5644]
### Dafny version

latest feat-rust

### Code to produce this issue

```dafny
module Spoo {

  newtype uint8 = x: int | 0 <= x < 0x100
  newtype uint16 = x: int | 0 <= x < 0x10000
  function method UInt16ToSeq(x: uint16): (ret: seq<uint8>)
    ensures |ret| == 2
    ensures 0x100 * ret[0] as uint16 + ret[1] as uint16 == x
  {
    var b0 := (x / 0x100) as uint8;
    var b1 := (x % 0x100) as uint8;
    [b0, b1]
  }

  method {:test} TestSetToOrderedSequenceManyItems() {
    var a := set x:uint16 | 0 <= x < 0xFFFF :: UInt16ToSeq(x);
  }
}
```

### Command to run and resulting output

```code
dafny -noVerify -compileTarget:rs -compile:0 -functionSyntax:3 -spillTargetCode:3 fail.dfy


(0,-1): Error: Microsoft.Dafny.UnsupportedInvalidOperationException: Expected ExprContainer, got Microsoft.Dafny.ConcreteSyntaxTree (ID: f_unsupported_feature)
```

### What happened?

Got the above error. Expected no error.

### What type of operating system are you experiencing the problem on?

Mac

**Comment by ajewellamz** [comment]
I get the same error message compiling
https://github.com/dafny-lang/libraries/blob/master/src/JSON/Grammar.dfy
triggered on line 58

**Comment by ssomayyajula** [comment]
Fix in `feat-rust` / 5227538
